### ISSUE TITLE
CloudWatch: AWS GetMetricData with 'Expressions' input always adds value of "${LABEL}"

### ISSUE THREAD

**Issue by salavessa** [grafana/grafana#51714]
**What happened**:
After updating grafana from `8.5.5` to `9.0.2` the `Label` override fields (`Alias` before grafana 9) for `Time series` and `Graph (old)` visualizations (using a `CloudWatch` datasource) are always getting the value of `${LABEL}` appended (unless it is specified in the `Label` expression itself).

**What you expected to happen**:
Have the value of `${LABEL}` automatically injected in relevant elements (e.g. labels and tooltips) only if the optional `Label` field is empty.

**How to reproduce it (as minimally and precisely as possible)**:
Create a `Time series` using `CloudWatch` datasource and set the `Label` value with a static value, e.g. `X` - the effective value will be rendered as `X ${LABEL}` instead of `X`.

![grafana9-cloudwatch-query-labels](https://user-images.githubusercontent.com/15344273/177161450-ef520e52-b59d-4147-961f-51a6e975fa64.png)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.0.2
- Data source type & version: CloudWatch
- OS Grafana is installed on: linux (docker)
- User OS & Browser: win10 & chrome 98
- Grafana plugins: n/a
- Others: official grafana docker image

**Comment by fridgepoet** [comment]
Thanks for reporting! 
I was able to reproduce the issue in Code builder mode only (same as in your screenshot).

**Environment:**
* Grafana version: 9.1.0-pre (main branch)
* OS Grafana is installed on: MacOS (local)

**Comment by fridgepoet** [comment]
(Forgot to mention that this doesn't happen in Builder mode, in case that is helpful as a temporary workaround.)

**Comment by salavessa** [comment]
Hi @fridgepoet,
Unfortunately a big part of the queries I have seem to be too complex to perform using the Builder mode and, as usual, I'm sure you'll get this sorted pretty quickly 🥇
Thanks!

**Comment by fridgepoet** [comment]
Hi again @salavessa, so what you reported is (undocumented I believe) behavior from the [GetMetricData api ](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html) that we're using in the CloudWatch plugin.

## TLDR
AWS is responding with the extra label, but only with the `Expression` input type. This is unexpected.

## Description of what we observed:
In the `GetMetricData` [query](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDataQuery.html), we either fill `Expression` **or** `MetricStat`.

In the CloudWatch plugin, we send input to `GetMetricData` as:
* `Expression`
  * Code mode (like your screenshot)
  * Builder mode with Dimensions with wildcard value `*`
* or `Metric Stat`
  * Builder mode without Dimensions with wildcard value `*`

For some reason, the `Expression` mode always appends the ${Label} to the return label value. 

### Examples with the aws-cli:

#### Query with `Expression`:
`aws cloudwatch get-metric-data --start-time 2022-06-01T00:00:00Z --end-time 2022-07-04T16:19:00Z --metric-data-queries file://expression.json`
where expression.json is:
```[
 {
   "Id": "e1",
   "Expression": "SEARCH(' {AWS/Lambda, FunctionName} MetricName=Invocations name-of-your-lambda-function', 'Sum', 3600)",
   "Label": "X"
 }
]
```
Response:
```
...
"Label": "X Invocations",
...
```
#### Query with `MetricStat`:
` aws cloudwatch get-metric-data --start-time 2022-06-01T00:00:00Z --end-time 2022-07-04T16:19:00Z --metric-data-queries file://metric-stat.json`
where metric-stat.json is:
```[
   {
      "Id":"e1",
      "MetricStat":{
         "Metric":{
            "Namespace":"AWS/Lambda",
            "MetricName":"Invocations",
            "Dimensions":[
               {
                  "Name":"FunctionName",
                  "Value":"name-of-your-lambda-function"
               }
            ]
         },
         "Period":3600,
         "Stat":"Sum"
      },
      "Label":"X"
   }
]
```
Response:
```
... 
"Label": "X",
...
```

**Comment by salavessa** [comment]
Hi @fridgepoet,

Thanks for the info and detailed explanation.
That's unfortunate and far from ideal - honestly I can't see AWS being quick/willing to change what seems an issue/feature on their side.
Is there any way to have grafana's 9.x query behavior for expressions to match the one in previous grafana versions which works as expected even with the unexpected/undocumented AWS behavior?

Thanks!

**Comment by fridgepoet** [comment]
Forgot to mention:
Instead of rolling back you can disable the use of dynamic labels in your conf `custom.ini`:
```
[feature_toggles]

# Use dynamic labels in CloudWatch datasource
cloudWatchDynamicLabels = false
```

**Comment by iwysiu** [comment]
It sounds like the problem has been resolved, so I'm closing this. Feel free to comment if there are further issues!

**Comment by salavessa** [comment]
Hi @iwysiu,

I'd say "my problem" has been resolved (through a workaround), but the actual issue reported in the issue is still there.

Not sure what the approach to this specific issue by grafana but as I see it:
- if grafana will just wait for AWS to change/fix the behavior on their side then I guess there's nothing left to see here (this can be closed and forgotten);
- if grafana isn't meant to have known issues out-of-the-box I think `cloudWatchDynamicLabels = false` should be the default, at least until the expected behavior happens;
- if grafana would develop a "dirty hack" to circumvent what seems to be an issue on AWS side, the this issue should be reopened.

As this is not currently having any impact in any of my grafana instances I'm ok with any of the above or some other approaches you may have on it.

Thanks!
